### PR TITLE
DIT-1050 add permission checks to "action" links 

### PR DESCRIPTION
**Pull request by SteveSugden**
This is part of the "not taking ownership" journey

### CODE REVIEW

File: test/unit/uk/gov/hmrc/tariffclassificationfrontend/views/partials/liabilities/LiabilityActionsViewSpec.scala [hunk 1]
```diff
@@ -37,10 +37,30 @@ class LiabilityActionsViewSpec extends ViewSpec {
       val doc = view(liability_actions(c, 0)(requestWithPermissions(Permission.SUSPEND_CASE), messages, appConfig))
 
       // Then
-      doc should containElementWithID("refer-case-button")
-      doc should containElementWithID("reject-case-button")
       doc should containElementWithID("suspend-case-button")
 
+      doc should not (containElementWithID("refer-case-button"))
```
Comment (ed0906) [inline test/unit/uk/gov/hmrc/tariffclassificationfrontend/views/partials/liabilities/LiabilityActionsViewSpec.scala]: no need for the extra brackets here

**Comment by SteveSugden** [comment]
I'm going to have another look at this.  Just realised that the heading "Actions on this case" should probably not be there if all links are hidden.

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/ShowActions.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+class ShowActions(caseStatus: CaseStatus)(implicit request: AuthenticatedRequest[_]) {
+
+  def refer: Boolean = caseStatus == CaseStatus.OPEN && request.hasPermission(Permission.REFER_CASE)
+  def reject: Boolean = caseStatus == CaseStatus.OPEN && request.hasPermission(Permission.REJECT_CASE)
+  def suspend: Boolean = caseStatus == CaseStatus.OPEN && request.hasPermission(Permission.SUSPEND_CASE)
+  def release: Boolean = caseStatus == CaseStatus.NEW && request.hasPermission(Permission.RELEASE_CASE)
+  def suppress: Boolean = caseStatus == CaseStatus.NEW && request.hasPermission(Permission.SUPPRESS_CASE)
+  def reopen: Boolean = (caseStatus == CaseStatus.SUSPENDED || caseStatus == CaseStatus.REFERRED) && request.hasPermission(Permission.REOPEN_CASE)
+
+  def any: Boolean = refer || reject || suspend || release || suppress || reopen
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/ShowActions.scala]: If you define an object in the same file
```
object ShowActions {
   def apply(caseStatus: CaseStatus): ShowActions = new ShowActions(caseStatus)
}
```

then later on when you want to use it you dont need the `new` keyword

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/ShowActions.scala]: I guess its only a win if we use the constructor a lott

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/ShowActions.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.views.partials
+
+import uk.gov.hmrc.tariffclassificationfrontend.models.CaseStatus.CaseStatus
+import uk.gov.hmrc.tariffclassificationfrontend.models.{CaseStatus, Permission}
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.AuthenticatedRequest
+
+class ShowActions(caseStatus: CaseStatus)(implicit request: AuthenticatedRequest[_]) {
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/ShowActions.scala]: Looks like a Unit test is missing

**Pull request merged**